code/script.js

```diff
@@ -255,59 +270,165 @@ const checkQuestion = () => {
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === 'hair') {
+      if (currentQuestion.value === secret.hair) {
+        keep = true
+        filterCharacters(keep)
+        console.log(keep)
+      } else {
+        keep  = false
+        filterCharacters(keep)
+        console.log(keep)
+      }
+  } else if (category === 'eyes') {
+    
+      if (currentQuestion.value === secret.eyes) {
+        keep = true
+        filterCharacters(keep)
+        console.log(keep)
+      } else {
+        keep  = false
+        filterCharacters(keep)
+        console.log(keep)
+      }
+  } else if (category === 'accessories') {
+    if (currentQuestion.value.includes('glasses')) {
+      if (currentQuestion.value.includes('glasses') === secret.accessories.includes('glasses')) {
+        keep = true
+        filterCharacters(keep)
+        console.log(keep)
+      } else if (currentQuestion.value.includes('glasses') !== secret.accessories.includes('glasses')) {
+        keep  = false
+        filterCharacters(keep)
+        console.log(keep)
+      }
+    } else if (currentQuestion.value.includes('hat')) {
+      if (currentQuestion.value.includes('hat') === secret.accessories.includes('hat')){
+        keep = true
+        filterCharacters(keep)
+        console.log(keep)
+      } else if(currentQuestion.value.includes('hat') !== secret.accessories.includes('hat')){
+        keep  = false
+        filterCharacters(keep)
+        console.log(keep)
+      }
+    }
+  } else if (category === 'other') {
+    if (currentQuestion.value === secret.other.join()) {
+      keep = true
+      filterCharacters(keep)
+      console.log(keep)
+    } else {
+      keep  = false
+      filterCharacters(keep)
+      console.log(keep)
+    }
   }
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
+  
   // Show the correct alert message for different categories
   if (category === 'accessories') {
-    if (keep) {
+    if (keep === true) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${value}! Keep all people that wears ${value}.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${value}! Remove all people that wears ${value}.`
       )
     }
   } else if (category === 'other') {
     // Similar to the one above
-  } else {
-    if (keep) {
+    if (keep === true) {
```

Comment:
If you use the comment from line 335, then you could save this step and just code everything together. 
if (category === 'color') {
    if (keep) {
      charactersInPlay = charactersInPlay.filter((person) => person [category].includes(value))
      alert(`Yes, the person is ${value}! Keep all the perople that are ${value}`)
    } else {
      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
      alert(`No, the person is not ${value}! Remove all the people that are ${value}`)